src/main/java/school/faang/BJS2_68051/User.java

```diff
@@ -0,0 +1,38 @@
+package school.faang.BJS2_68051;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode
+@ToString
+public class User {
+
+    private int id;
+    private String name;
+    private int age;
+    private Set<String> userActivities;
+
+    public static Map<User, String> findHobbyLovers(List<User> userList, List<String> activity) {
+        return userList.stream().filter(user -> user.getUserActivities().stream().anyMatch(activity::contains))
```

Comment:
За стримы конечно лайк. Ты и сам заметил, что стрим как-то не получилось у тебя завершить. Когда нужно что-то найти, но при этом не потерять контекст, в частности найдунную активити, то можно использовать не filter, а map/flatMap:

 ```suggestion
        public static Map<User, String> findHobbyLovers(List<User> userList, List<String> activity) {
        return userList.stream()
                .flatMap(user -> user.getUserActivities() // не теряем юзера
                        .stream()
                        .filter(activity::contains)
                        .findFirst()
                        .map(matchedActivity -> Map.entry(user, matchedActivity)) // прихраниваем юзера вместе с найденной activity
                        .stream()
                ).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));

    }
```